### PR TITLE
[AD-997] Add tests to ensure SQL or mongo injection is not possible.

### PR DESCRIPTION
**Pull request by birschick-bq**
### Summary

[AD-997] Add tests to ensure SQL or mongo injection is not possible.

### Description

Add
- Tests for mongo injection
- Tests for SQL injection

### Related Issue

https://bitquill.atlassian.net/browse/AD-997

### Additional Reviewers
@affonsoBQ
@alinaliBQ
@andiem-bq
@alexey-temnikov
@birschick-bq

### CODE REVIEW

File: src/test/java/software/amazon/documentdb/jdbc/query/limitations/DocumentDbSqlInjectionTest.java [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+    @BeforeAll
+    void beforeAll() throws SQLException {
+        final BsonDocument document =
+                BsonDocument.parse(
+                        "{ \"_id\" : \"key\", \"array\" : [ "
+                                + "{ \"field\" : 1, \"field1\": \"value\" }, "
+                                + "{ \"field\" : 2, \"field2\" : \"value\" , \"field3\" : { \"field4\": 3} } ]}");
+        final BsonDocument otherDocument =
+                BsonDocument.parse(
+                        "{ \"_id\" : \"key1\", \"otherArray\" : [ { \"field\" : 1, \"field3\": \"value\" }, { \"field\" : 2, \"field3\" : \"value\" } ]}");
```
Comment (alinaliBQ) [inline src/test/java/software/amazon/documentdb/jdbc/query/limitations/DocumentDbSqlInjectionTest.java]: nit - formatting issue?

File: src/test/java/software/amazon/documentdb/jdbc/query/limitations/DocumentDbSqlInjectionTest.java [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+                "field",
+                "field1",
+                getDatabaseName(),
+                COLLECTION_NAME + "_array",
+                new StringBuilder(injection).reverse());
+        final DocumentDbMqlQueryContext queryContext5 = queryMapper.get(query5);
+        Assertions.assertNotNull(queryContext5);
+        final List<Bson> aggregateOperations5 = queryContext5.getAggregateOperations();
+        assertKeyNotExists(expectedKey, aggregateOperations5);
+        assertValueExists(injection, aggregateOperations5);
```
Comment (alinaliBQ) [inline src/test/java/software/amazon/documentdb/jdbc/query/limitations/DocumentDbSqlInjectionTest.java:144]: I'm confused here, if we're making sure Mongo injection is not possible, then why are we making sure that the injection exists in the aggregate options (i.e., by calling `assertValueExists(injection, <aggregateOptionsVar> )`)?

Comment (birschick-bq) [reply src/test/java/software/amazon/documentdb/jdbc/query/limitations/DocumentDbSqlInjectionTest.java:144]: @alinaliBQ 
The point of this assertion is that the injection has been interpreted only as a value, as is. We are ensuring the (attempted) injection despite quoting will only be interpreted as a value and nothing else.

File: src/test/java/software/amazon/documentdb/jdbc/query/limitations/DocumentDbSqlInjectionTest.java (hunk 1 above)
Comment (birschick-bq) [reply src/test/java/software/amazon/documentdb/jdbc/query/limitations/DocumentDbSqlInjectionTest.java]: It was copy/pasted - but formatted a more compact version.

File: src/test/java/software/amazon/documentdb/jdbc/query/limitations/DocumentDbSqlInjectionTest.java (hunk 2 above)
Comment (alinaliBQ) [reply src/test/java/software/amazon/documentdb/jdbc/query/limitations/DocumentDbSqlInjectionTest.java:144]: Oh I see, thanks for explaining!

**Review by alinaliBQ — COMMENTED** [review]
Aside from the build errors which do not seem to be caused by the new changes in the PR, the PR lgtm!

**Review by andiemontoyeah — APPROVED** [review]
Change LGTM!

**Pull request merged**